daffodil-tdml-processor/src/main/scala/org/apache/daffodil/tdml/processor/runtime2/Runtime2TDMLDFDLProcessor.scala

```diff
@@ -0,0 +1,234 @@
+      val compileResult = if (pf.isError) {
+        Left(pf.getDiagnostics) // C code compilation diagnostics
+      } else {
+        val dp = new Runtime2TDMLDFDLProcessor(generatedCodeCompiler)
+        Right((pf.getDiagnostics, dp))
+      }
+      compileResult
+    }
+    res
+  }
+
+}
+
+/**
+ * Delegates all execution, error gathering, error access to the Runtime2DataProcessor object.
+ * The responsibility of this class is just for TDML matching up. That is dealing with TDML
+ * XML Infosets, feeding to the unparser, creating XML from the result created by the
+ * Runtime2DataProcessor object. All the "real work" is done by generatedCodeCompiler.dataProcessor.
+ */
+class Runtime2TDMLDFDLProcessor(generatedCodeCompiler: GeneratedCodeCompiler) extends TDMLDFDLProcessor {
+
+  override type R = Runtime2TDMLDFDLProcessor
+
+  private val dataProcessor = generatedCodeCompiler.dataProcessor
+  private var anyErrors: Boolean = false
+  private var diagnostics: Seq[Diagnostic] = Nil
+
+  @deprecated("Use withDebugging.", "2.6.0")
+  override def setDebugging(b: Boolean) = ???
+  override def withDebugging(b: Boolean): Runtime2TDMLDFDLProcessor = this
+
+  @deprecated("Use withTracing.", "2.6.0")
+  override def setTracing(bool: Boolean): Unit = ???
+  override def withTracing(bool: Boolean): Runtime2TDMLDFDLProcessor = this
+
+  @deprecated("Use withDebugger.", "2.6.0")
+  override def setDebugger(db: AnyRef): Unit = ???
+  override def withDebugger(db: AnyRef): Runtime2TDMLDFDLProcessor = this
+
+  @deprecated("Use withValidationMode.", "2.6.0")
+  override def setValidationMode(validationMode: ValidationMode.Type): Unit = ???
+  override def withValidationMode(validationMode: ValidationMode.Type): Runtime2TDMLDFDLProcessor = this
+
+  @deprecated("Use withExternalDFDLVariables.", "2.6.0")
+  override def setExternalDFDLVariables(externalVarBindings: Seq[Binding]): Unit = ???
+  override def withExternalDFDLVariables(externalVarBindings: Seq[Binding]): Runtime2TDMLDFDLProcessor = this
+
+  // Actually run the C code and save any errors to be returned here
+  override def isError: Boolean = anyErrors
+  override def getDiagnostics: Seq[Diagnostic] = diagnostics
+
+  // This part will change a lot (it will execute C code instead).
+  // Whatever the parse produces needs to be converted into XML for comparison.
+  // We'll need a way to convert, say, a C struct to XML, and XML to C struct.
+  // The C code will need a bunch of toXML methods so it can produce output
+  // for comparison.
+  override def parse(is: java.io.InputStream, lengthLimitInBits: Long): TDMLParseResult = {
+    // We will run the generated and compiled C code, collect and save any errors
+    // and diagnostics to be returned in isError and getDiagnostics, and build an
+    // infoset.  Our context here is TDML-related, so we need to move that functionality
+    // to something generic that we call from here, you're saying.  I got it.  So we
```

Comment:
Clean up conversational comments. Much of this discussion suggests uncertainty that I think is much clearer now.